Model/Data/OrderUpdate.php

```diff
@@ -0,0 +1,192 @@
+<?php
+declare(strict_types=1);
+
+namespace Amwal\Payments\Model\Data;
+
+use Amwal\Payments\Model\Checkout\AmwalCheckoutAction;
+use Amwal\Payments\Plugin\Sentry\SentryExceptionReport;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\RequestOptions;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Amwal\Payments\Model\Config;
+use Amwal\Payments\Model\GetAmwalOrderData;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Mail\MessageInterface;
+use Magento\Framework\Mail\TransportInterfaceFactory;
+use Magento\Sales\Model\Order;
+use Magento\Sales\Model\OrderNotifier;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Amwal\Payments\Model\Checkout\InvoiceOrder;
+use Psr\Log\LoggerInterface;
+use Amwal\Payments\Model\AmwalClientFactory;
+
+class OrderUpdate
+{
+    private OrderRepositoryInterface $orderRepository;
+    private StoreManagerInterface $storeManager;
+    private GetAmwalOrderData $getAmwalOrderData;
+    private Config $config;
+    private OrderNotifier $orderNotifier;
+    private TransportInterfaceFactory $transportFactory;
+    private MessageInterface $message;
+    private ScopeConfigInterface $scopeConfig;
+    private InvoiceOrder $invoiceAmwalOrder;
+    private LoggerInterface $logger;
+    private AmwalClientFactory $amwalClientFactory;
+    private SentryExceptionReport $sentryExceptionReportr;
+
+    public function __construct(
+        OrderRepositoryInterface  $orderRepository,
+        StoreManagerInterface     $storeManager,
+        GetAmwalOrderData         $getAmwalOrderData,
+        Config                    $config,
+        OrderNotifier             $orderNotifier,
+        TransportInterfaceFactory $transportFactory,
+        MessageInterface          $message,
+        ScopeConfigInterface      $scopeConfig,
+        InvoiceOrder              $invoiceAmwalOrder,
+        LoggerInterface           $logger,
+        AmwalClientFactory        $amwalClientFactory,
+        SentryExceptionReport     $sentryExceptionReportr
+    )
+    {
+        $this->orderRepository = $orderRepository;
+        $this->storeManager = $storeManager;
+        $this->getAmwalOrderData = $getAmwalOrderData;
+        $this->config = $config;
+        $this->orderNotifier = $orderNotifier;
+        $this->transportFactory = $transportFactory;
+        $this->message = $message;
+        $this->scopeConfig = $scopeConfig;
+        $this->invoiceAmwalOrder = $invoiceAmwalOrder;
+        $this->logger = $logger;
+        $this->amwalClientFactory = $amwalClientFactory;
+        $this->sentryExceptionReport = $sentryExceptionReportr;
+    }
+
+    /*
+     * @param OrderRepositoryInterface $orderRepository
+     * @param getAmwalOrderData $amwalOrderData
+     * @param string $historyComment
+     * return bool
+     */
+    public function update($order, $amwalOrderData, $historyComment = '')
```

Comment:
since $order and $amwalOrderData are provided externally, can we provide an assertion that the transaction_id of both matches?